string_parser.hpp

```diff
@@ -55,7 +54,7 @@ std::istream& operator>>(std::istream& is, string_parser& parser);
 
 class string_parser {
 public:
-    constexpr string_parser() noexcept : data_{}, pos_{0} {}
+    string_parser() noexcept : data_{}, pos_{0} {}
```

Comment:
Am curious on the reason for removal of the constexpr here?

Reply:
I had some compilation problems with them, don't remember from which compiler. Removing the constexpr made the code build, and I never thought about it again. I'm not incredibly versed in the correct use of this keyword, but if you want, I'll find out what my issue was, and if there's a better solution.

Reply:
If so, we might need to figure something out.